### ISSUE TITLE
Block-scoped variable used before its declaration

### ISSUE THREAD

**Issue by fnatte** [anymaniax/orval#244]
## What are the steps to reproduce this issue?

Generate client with `mode: 'single'` using spec:
```yaml
openapi: '3.0.0'
info:
  version: 1.0.0
  title: Test
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /colors:
    get:
      summary: List colors
      operationId: listColors
      tags:
        - colors
      responses:
        '200':
          description: An array of colors
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ColorContainer'
components:
  schemas:
    ColorContainer:
      type: object
      properties:
        color:
          allOf:
            - $ref: '#/components/schemas/Color'
    Color:
      type: string
      enum:
        - red
        - green
        - blue

```

## What happens?

The result:
```typescript
export type ColorContainerColor = Color

export const ColorContainerColor = {...Color};

export interface ColorContainer {
  color?: ColorContainerColor;
}

export type Color = 'red' | 'green' | 'blue';
```

contains invalid TypeScript with error `Block-scoped variable 'Color' used before its declaration. ts(2448)`

## What were you expecting to happen?

I would expect type `Color` to be defined before `ColorContainerColor`

## Any other comments?

Not sure how to tackle this one. Maybe some kind of dependency data structure needs to be built so that the schemas can be ordered accordingly before being written.

Two workarounds:
- order the schemas by hand in a transformer
- split model files

## What versions are you using?

49290694a8e072ee4f7311c0a04ee92da9cfef35

**Comment by anymaniax** [comment]
should be fixed with the next release

**Comment by anymaniax** [comment]
Should be fixed with `6.4.0` can you test when you have time, please?

**Comment by anymaniax** [comment]
Closing this issue since it's normally fixed. Don't hesitate if you have any problem
